### PR TITLE
Deprecate HTTP(S)

### PR DESCRIPTION
**Pull request by jaspervdm**
Implementation of the [Deprecate HTTP(S) Transactions RFC](https://github.com/mimblewimble/grin-rfcs/pull/54).

### CODE REVIEW

File: impls/src/adapters/mod.rs [hunk 1]
```diff
@@ -59,64 +58,3 @@ pub trait SlateGetter {
...
-				return Err(
-					ErrorKind::WalletComms("Tor Configuration required".to_string()).into(),
-				);
-			}
-			Some(tc) => Box::new(
-				HttpSlateSender::with_socks_proxy(&dest, &tc.socks_proxy_addr, &tc.send_config_dir)
-					.map_err(|_| invalid())?,
-			),
-		},
-		"keybase" => Box::new(KeybaseChannel::new(dest)?),
```
Comment (j01tz) [inline impls/src/adapters/mod.rs:101]: With this change, is there any reason to import or keep around any of the keybase logic in adapters?

**Review by j01tz — APPROVED** [review]
Looks good overall, thanks for taking care of this @jaspervdm.

I ran it on a test instance with Tor not installed and got following output

20201124 09:45:40.939 WARN grin_wallet_controller::controller - Starting TOR Hidden Service for API listener at address rglop53gkfop22lfka5z2juxsv4f6uciwseg7dslsay72otevsrsupyd, binding to 127.0.0.1:13415
20201124 09:45:40.944 WARN grin_wallet_controller::controller - Unable to start TOR listener; Check that TOR executable is installed and on your path
20201124 09:45:40.944 WARN grin_wallet_controller::controller - Tor Error: Tor Process Error: Process("TOR executable (`tor`) not found. Please ensure TOR is installed and on the path: Os { code: 2, kind: NotFound, message: \"No such file or directory\" }")
20201124 09:45:40.944 WARN grin_wallet_controller::controller - Listener will be available via HTTP only
20201124 09:45:40.945 WARN grin_wallet_controller::controller - Starting HTTP Foreign listener API server at 127.0.0.1:13415.
20201124 09:45:40.945 WARN grin_wallet_controller::controller - HTTP Foreign listener started.
I think it is doing what we expect but a couple of usability things that may be worth noting for a future PR:

- do we want the first log line above to return a tor address or a slatepack address? this could add confusion to return a raw tor address (or am I missing something in how the API is done?)
- even though http is no longer supported, the warning logs can confuse a user into thinking http is still working

**Review by antiochp — APPROVED** [review]
Looks good to me. :+1:

File: impls/src/adapters/mod.rs (hunk 1 above)
Comment (jaspervdm) [reply impls/src/adapters/mod.rs:101]: Afaik we still support Keybase as a method. This particular block of code was just unused so I got rid of it entirely.

**Comment by jaspervdm** [comment]
@j01tz 
(i edited your comment by accident instead of replying to it, my bad)

> even though http is no longer supported, the warning logs can confuse a user into thinking http is still working

Technically we still have a HTTP listener, it just runs locally. This listener is also used by the TOR hidden service. But I do agree we could improve on the text here in a future PR.

**Pull request merged**